lib/babe/build.go

```diff
@@ -66,7 +67,7 @@ type BlockBuilder struct {
 // NewBlockBuilder creates a new block builder.
 func NewBlockBuilder(kp *sr25519.Keypair, ts TransactionState,
 	bs BlockState, proof *VrfOutputAndProof,
-	authidx uint32) (*BlockBuilder, error) {
+	authidx uint32, slotToIfPrimary map[uint64]bool) (*BlockBuilder, error) {
```

Comment:
there's actually two types of secondary production (VRF and Plain) so it would be nice to make this more generic. another idea is to pass the `PreRuntimeDigest` into this function and create it at the caller level, which would make more sense since creating the pre-runtime digest doesn't actually depend on anything in the `BlockBuilder`